internal/pkg/cli/local_run.go

```diff
@@ -152,21 +149,35 @@ func newLocalRunOpts(vars localRunVars) (*localRunOpts, error) {
 		o.repository = repository.NewWithURI(ecr.New(defaultSessEnvRegion), repoName, resources.RepositoryURLs[o.wkldName])
 		return nil
 	}
-	opts.buildContainerImages = func(o *localRunOpts) error {
-		gitShortCommit := imageTagFromGit(o.cmd)
+	opts.buildContainerImages = func(mft manifest.DynamicWorkload) (map[string]string, error) {
+		gitShortCommit := imageTagFromGit(opts.cmd)
 		image := clideploy.ContainerImageIdentifier{
 			GitShortCommitTag: gitShortCommit,
 		}
-		return clideploy.BuildContainerImages(&clideploy.ImageActionInput{
-			Name:               o.wkldName,
-			WorkspacePath:      o.ws.Path(),
+		out := &clideploy.UploadArtifactsOutput{}
+		if err := clideploy.BuildContainerImages(&clideploy.ImageActionInput{
+			Name:               opts.wkldName,
```

Comment:
I'm worried that `opts.wkldName` will not have the proper value. I'm probably wrong, but I _think_ that `opts.wkldName` will probably be evaluated when the function is declared, not called. `opts.repository` is probably not an issue because it is a pointer - not sure about `opts.repository.Login` though.

Reply:
i just verified to ensure this isn't happening, and it looks like it's working to me! 😊 
![Screenshot 2023-08-24 at 10 35 28](https://github.com/aws/copilot-cli/assets/10566468/6f5fec69-b513-4888-9947-e5ef79150b06)